tests/director/tests/formation_api_test.go

```diff
@@ -18,6 +18,72 @@ import (
 const unassignFormationCategory = "unassign formation"
 
+func TestGetFormation(t *testing.T) {
+	ctx := context.Background()
+	formationName := "formation1"
+
+	t.Logf("Should create formation: %q", formationName)
+	formation := fixtures.CreateFormation(t, ctx, certSecuredGraphQLClient, formationName)
+	require.Equal(t, formationName, formation.Name)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, formationName)
+
+	t.Logf("Should get formation %q by id %q", formationName, formation.ID)
+	var gotFormation graphql.Formation
+	getFormationReq := fixtures.FixGetFormationRequest(formation.ID)
+	err := testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, getFormationReq, &gotFormation)
+	require.NoError(t, err)
+	require.Equal(t, formation, gotFormation)
+
+	t.Logf("Should delete formation %q", formationName)
+	deleteFormation := fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, formationName)
+	assert.Equal(t, formation, *deleteFormation)
+
+	t.Logf("Should NOT get formation %q by id %q because it is already deleted", formationName, formation.ID)
+	var nonexistentFormation *graphql.Formation
+	getNonexistentFormationReq := fixtures.FixGetFormationRequest(formation.ID)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, getNonexistentFormationReq, nonexistentFormation)
+	require.Error(t, err)
+	require.Nil(t, nonexistentFormation)
+}
+
+func TestListFormations(t *testing.T) {
+	// Pre-cleanup because the formations table may be dirty by previous tests in the director,
+	// which delete their created formations after the end of all director tests.
+	tenantID := tenant.TestTenants.GetDefaultTenantID()
+	tenant.TestTenants.CleanupTenant(tenantID)
+
+	ctx := context.Background()
+
+	firstFormationName := "formation1"
+	secondFormationName := "formation2"
+
+	first := 100
+
+	expectedFormations := 0
+	t.Logf("List should return %d formations", expectedFormations)
+	formationPage1 := fixtures.ListFormations(t, ctx, certSecuredGraphQLClient, first, expectedFormations)
+	require.NotNil(t, formationPage1)
+	require.Equal(t, expectedFormations, formationPage1.TotalCount)
+
+	t.Logf("Should create formation: %q", firstFormationName)
+	firstFormation := fixtures.CreateFormation(t, ctx, certSecuredGraphQLClient, firstFormationName)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, firstFormationName)
+
+	t.Logf("Should create formation: %q", secondFormationName)
+	secondFormation := fixtures.CreateFormation(t, ctx, certSecuredGraphQLClient, secondFormationName)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, secondFormationName)
+
+	expectedFormations = 2
+	t.Logf("List should return %d formations", expectedFormations)
+	formationPage2 := fixtures.ListFormations(t, ctx, certSecuredGraphQLClient, first, expectedFormations)
+	require.NotNil(t, formationPage2)
+	assert.Equal(t, expectedFormations, formationPage2.TotalCount)
+	assert.Subset(t, formationPage2.Data, []*graphql.Formation{
```

Comment:
I think it would be better to use `assert.ElementsMatch(t, expected, actual)` instead of `assert.Subset()` because the `Subset()` doesn't check that the number of elements in both arrays is equal while `ElementsMatch()` does:

`ElementsMatch asserts that the specified listA(array, slice...) is equal to specified listB(array, slice...)`

For example, if we use `Subset()` and we expect only `firstFormation` and `secondFormation` to be present in `formationPage2.Data` but in `formationPage2.Data` there are X more formations this assert will pass because the two formations will be indeed present but we want  to be sure that only the two are returned.

Reply:
Actually, on the line above(80) we check if their count is the same. And if their count is the same and one of the sets is a subset of the other one => they are equal, but I will change it because it's a little bit misleading.